src/starkware_utils/commitment_tree/patricia_tree/patricia_guess_descents.rs

```diff
@@ -0,0 +1,475 @@
+        let height = Height(3);
+
+        // The update tree should look like this:
+        //        r
+        //    1       0
+        //  1   0   0   0
+        // 0 u 0 0 0 0 0 0
+        // Resulting in a descent of length 3.
+        let update_tree = build_update_tree(height, vec![(TreeIndex::from(1u64), LF::new(Felt252::from(128)))]);
+
+        // Start from an empty tree
+        let prev_root = BigUint::from(0u64);
+        // Setting an empty tree as end state works for testing the function.
+        // The descent map will then only depend on the structure of the update tree.
+        // Of course this does not make any sense in a real-setting.
+        let new_root = BigUint::from(0u64);
+
+        let preimage = Preimage::new();
+
+        let descent_map = patricia_guess_descents::<LF>(height, update_tree, &preimage, prev_root, new_root).unwrap();
+        print_descent_map(&descent_map);
+        assert_eq!(
+            descent_map,
+            DescentMap::from([((Felt252::from(3), Felt252::from(0)), vec![Felt252::from(3), Felt252::from(1)])]),
+        );
+    }
+
+    /// Tests generating a descent map for an update tree of two adjacent leaves, against
+    /// an empty tree. The new tree is also empty to have the descent map reflect
+    /// the structure of the update tree and nothing else.
+    #[rstest]
+    fn test_guess_descents_update_two_adjacent_leaves() {
+        let height = Height(3);
+
+        // The update tree should look like this:
+        //        r
+        //    1       0
+        //  1   0   0   0
+        // u u 0 0 0 0 0 0
+        // Resulting in a descent of length 2.
+        let update_tree = build_update_tree(
+            height,
+            vec![
+                (TreeIndex::from(0u64), LF::new(Felt252::from(127))),
+                (TreeIndex::from(1u64), LF::new(Felt252::from(128))),
+            ],
+        );
+
+        // Start from an empty tree
+        let prev_root = BigUint::from(0u64);
+        // Setting an empty tree as end state works for testing the function.
+        // The descent map will then only depend on the structure of the update tree.
+        // Of course this does not make any sense in a real-setting.
+        let new_root = BigUint::from(0u64);
+
+        let preimage = Preimage::new();
+
+        let descent_map = patricia_guess_descents::<LF>(height, update_tree, &preimage, prev_root, new_root).unwrap();
+        print_descent_map(&descent_map);
+        assert_eq!(
+            descent_map,
+            DescentMap::from([((Felt252::from(3), Felt252::from(0)), vec![Felt252::from(2), Felt252::from(0)])]),
+        );
+    }
+
+    /// Tests generating a descent map for an update tree of two leaves, against
+    /// an empty tree. The new tree is also empty to have the descent map reflect
+    /// the structure of the update tree and nothing else.
+    #[rstest]
+    fn test_guess_descents_update_two_leaves() {
+        let height = Height(3);
```

Comment:
The tests only use a height of 3, which I think fails to cover some other important cases. For example, a tree that requires two descents of different lengths.

The "real world" will have a static height of `251` unless I'm missing something. I suggest at least adding a test case with a height on this order to flush out any absurd cases like OOM, etc.

Reply:
Yeah that's probably not a bad idea! I used height 3 because it's easy to reason with.